tests/unit/meta/test_snap_yaml.py

```diff
@@ -124,6 +124,70 @@ def test_assumes(simple_project, new_dir):
     )
 
 
+def test_build_base_devel(simple_project, new_dir):
+    """Devel base return devel grade and no build-base in snap.yaml."""
+    snap_yaml.write(
+        simple_project(build_base="devel"),
+        prime_dir=Path(new_dir),
+        arch="amd64",
+    )
+    yaml_file = Path("meta/snap.yaml")
+    assert yaml_file.is_file()
+
+    content = yaml_file.read_text()
+    assert content == textwrap.dedent(
+        """\
+        name: mytest
+        version: 1.29.3
+        summary: Single-line elevator pitch for your amazing snap
+        description: test-description
+        architectures:
+        - amd64
+        base: core22
+        apps:
+          app1:
+            command: bin/mytest
+        confinement: strict
+        grade: devel
+        environment:
+          LD_LIBRARY_PATH: ${SNAP_LIBRARY_PATH}${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}
+          PATH: $SNAP/usr/sbin:$SNAP/usr/bin:$SNAP/sbin:$SNAP/bin:$PATH
+        """
+    )
+
+
+def test_build_base_stable(simple_project, new_dir):
+    """Stable base return stable grade and no build-base in snap.yaml."""
+    snap_yaml.write(
+        simple_project(build_base="core22"),
```

Comment:
We're expecting the output to be the same regardless of the build base, right? Should we parametrize this as a single test?

Reply:
I thought about it, but the grade changes, this makes it easier to review